libcudacxx/test/libcxx/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+
+set(LIBCXX_LIT_VARIANT "libcxx" CACHE STRING
+    "Configuration variant to use for LIT.")
+
+set(LIBCXX_TEST_LINKER_FLAGS "" CACHE STRING
+    "Additonal linker flags to pass when compiling the tests")
+
+# <nv/target> is found in the parent project, it is required in all headers now
+set(LIBCXX_TEST_COMPILER_FLAGS
+    "-I${libcudacxx_SOURCE_DIR}/include" CACHE STRING
+    "Additonal compiler flags to pass when compiling the tests")
+
+# sccache cannot handle the -Fd option generationg pdb files
+if (MSVC)
+  set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT Embedded)
+endif()
+
+# Ensure we catch all warnings
+if (MSVC)
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS "${LIBCUDACXX_TEST_COMPILER_FLAGS} /W4 /WX")
+else()
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS "${LIBCUDACXX_TEST_COMPILER_FLAGS} -Wall -Wextra")
+endif()
+
+# Intel OneAPI compiler has fast math enabled by default which breaks almost all floating point tests
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM" OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS "${LIBCUDACXX_TEST_COMPILER_FLAGS} -fno-fast-math")
+endif()
+
+include(AddLLVM)
+
+set(LIBCUDACXX_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}")
```

Comment:
[<PROJ_NAME>_BINARY_DIR](https://cmake.org/cmake/help/latest/variable/PROJECT-NAME_BINARY_DIR.html) is defined by CMake and expected to be the top-level binary dir for a project. Probably want to use something else here to avoid surprises later.